encodings/fastlanes/src/bitpacking/compute/filter.rs

```diff
@@ -110,6 +104,36 @@ fn filter_indices<T: NativePType + BitPacking + ArrowNativeType>(
     values
 }
 
+pub fn chunked_indices<F: FnMut(usize, &[usize])>(
+    mut indices: impl Iterator<Item = usize>,
+    offset: usize,
+    mut chunk_fn: F,
+) {
+    let mut indices_within_chunk: Vec<usize> = Vec::with_capacity(1024);
```

Comment:
might as well allocate after the early exit that immediately follows. also, consider putting this on the stack which may be faster? (obviously, we can have more than 1024 since take allows duplicates, but it's reasonable to then just treat more than 1024 as multiple chunks)